esp-hal/src/timer/timg.rs

```diff
@@ -303,6 +303,54 @@ where
 {
     /// Construct a new instance of [`TimerGroup`] in asynchronous mode
     pub fn new_async(_timer_group: impl Peripheral<P = T> + 'd) -> Self {
+        use crate::timer::timg::asynch::timg0_timer0_handler;
```

Comment:
Please move these imports closer to where they are used, I wouldn't mind reducing the number of `#[cfg]`s here.